outlines/base.py

```diff
@@ -12,11 +12,19 @@
     _update_dim_sizes,
 )
 
-# Allow nested loops, useful to run in notebooks
-try:
-    nest_asyncio.apply()
-except ValueError as e:
-    print("Could not apply nest_asyncio:", e)
+# Allow nested loops for running in notebook. We don't enable it globally as it
+# may interfere with other libraries that use asyncio.
+if hasattr(builtins, "__IPYTHON__"):
+    try:
+        import nest_asyncio
+
+        nest_asyncio.apply()
+    except ImportError as e:
+        print(
+            "Couldn't patch nest_asyncio because it's not installed. Running in the notebook might be have issues"
+        )
+    except ValueError as e:
+        print("Could not apply nest_asyncio:", e)
```

Comment:
When is `ValueError` raised?

Reply:
It was there from https://github.com/outlines-dev/outlines/pull/481 . Presumably some vllm integration but I didn't run into it myself

Reply:
We can remove it